ci/env/install-dependencies.sh

```diff
@@ -160,6 +160,9 @@ install_miniconda() {
     )
   fi
 
+  # Install mpi4py
+  "${WORKSPACE_DIR}"/ci/suppress_output conda install -c anaconda mpi4py -y
```

Comment:
Hmm it is a bit weird we have it here? (it will be requested by every dev to download mpi4py)

Why don't we just make it test requirement?

Reply:
The problem is that only conda can install it. It has system deps and doesn't work with py311 :(